gax/src/gax.ts

```diff
@@ -209,33 +262,26 @@ export class CallSettings {
     let longrunning = this.longrunning;
     let apiName = this.apiName;
     let retryRequestOptions = this.retryRequestOptions;
+
+    // If the user provides a timeout to the method, that timeout value will be used
+    // to override the backoff settings.
+    if ('timeout' in options) {
+      timeout = options.timeout!;
+    }
     // If a method-specific timeout is set in the service config, and the retry codes for that
     // method are non-null, then that timeout value will be used to
     // override backoff settings.
     if (
-      retry !== undefined &&
-      retry !== null &&
-      retry.retryCodes !== null &&
-      retry.retryCodes.length > 0
+      retry?.retryCodesOrShouldRetryFn &&
+      ((isRetryCodes(retry!.retryCodesOrShouldRetryFn) &&
```

Comment:
Why wouldn't just `retry?.retryCodesOrShouldRetryFn` work, since it seems like we should enter this if it's a non-empty list of retry codes or a function?

Reply:
LOL it definitely does. This is a "i stared at this too long" mistake on my part.